src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySink.java

```diff
@@ -103,15 +105,21 @@ public void initialize(BatchRuntimeContext context) throws Exception {
   }
 
   @Override
-  public void transform(StructuredRecord input, Emitter<KeyValue<JsonObject, NullWritable>> emitter) {
-    JsonObject object = new JsonObject();
-    for (Schema.Field recordField : Objects.requireNonNull(input.getSchema().getFields())) {
-      // From all the fields in input record, decode only those fields that are present in output schema
-      if (schema.getField(recordField.getName()) != null) {
-        decodeSimpleTypes(object, recordField.getName(), input);
+  public void transform(StructuredRecord input, Emitter<KeyValue<Text, NullWritable>> emitter) {
+    StringWriter strWriter = new StringWriter();
+    try (JsonWriter writer = new JsonWriter(strWriter)) {
+      writer.beginObject();
```

Comment:
I assume structs are coming in a later PR then? Otherwise we could just call the method to write a struct here.

Reply:
structs are not part of this PR.